### PR TITLE
Feature/docfx gen

### PR DESCRIPTION
**Pull request by thalbern**
Overview
--- Added docfx support for documentation

Changes
--- added build config json file for docfx generation
--- added toc.yml files for creating navigation entries for docfx generated html website
--- moved all images used in documentation into External/ReadMeImages folders
--- fixed all broken/dead links in documentation 
--- changed all paths / links in documentation to relative paths so docfx can properly generate html links and do a link verification

### CODE REVIEW

File: api_doc/toc.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+- name: Conceptual
+  items: 
+  - name: Spatial Awareness
+    href: ../Documentation/Architecture/SpatialAwareness/toc.yml
+  - name: SDK
+    href: ../Assets/MixedRealityToolkit.SDK/toc.yml
+  - name: Tookit
```
Comment (MenelvagorMilsom) [inline api_doc/toc.yml]: Typo: _Too**l**kit_

File: docfx.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+  "metadata": [
+    {
+      "src": [
+        {
+          "files": [ "**/*.sln", "**/*.cs" ],
+          "exclude": [ "**/bin/**", "**/obj/**" ],
+          "src": "Assets/MixedRealityToolkit"
```
Comment (MenelvagorMilsom) [inline docfx.json]: Should we not include other folders for generated docs (e.g. MixedRealityToolkit.Providers, MixedRealityToolkit.SDK or MixedRealityToolkit.Services)

File: api_doc/toc.yml (hunk 1 above)
Comment (thalbern) [reply api_doc/toc.yml]: fixed typo

File: docfx.json (hunk 2 above)
Comment (thalbern) [reply docfx.json]: included all source folders for now

**Comment by chrisfromwork** [comment]
I don't have full context on how to validate these changes but am on board with these improvements. I also imagine that none of the changes here directly impact unity code/development environments.

If there is anything in particular that you would suggest when validating documentation changes, we should probably get it added to the wiki or on the documentation portal.

File: toc.yml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+- name: Home
+  href: README.md
+- name: Guides
+  href: Documentation/
+  homepage: Documentation/GettingStartedWithTheMRTK.md
```
Comment (chrisfromwork) [inline toc.yml]: Would we prefer our home page to be the github landing page readme.md compared to getting started with the MRTK?

**Comment by thalbern** [comment]
> I don't have full context on how to validate these changes but am on board with these improvements. I also imagine that none of the changes here directly impact unity code/development environments.
> 
> If there is anything in particular that you would suggest when validating documentation changes, we should probably get it added to the wiki or on the documentation portal.

As these are mostly config files there's not a lot you can look out for. Looking out for typos like Niall did is pretty much it - links are being checked by docfx, so there would be a warning if I put a broken link in one of the toc files or md files.

**Comment by chrisfromwork** [comment]
> > I don't have full context on how to validate these changes but am on board with these improvements. I also imagine that none of the changes here directly impact unity code/development environments.
> > If there is anything in particular that you would suggest when validating documentation changes, we should probably get it added to the wiki or on the documentation portal.
> 
> As these are mostly config files there's not a lot you can look out for. Looking out for typos like Niall did is pretty much it - links are being checked by docfx, so there would be a warning if I put a broken link in one of the toc files or md files.

Is there a mechanism for running docfx building locally?

**Comment by thalbern** [comment]
> > > I don't have full context on how to validate these changes but am on board with these improvements. I also imagine that none of the changes here directly impact unity code/development environments.
> > > If there is anything in particular that you would suggest when validating documentation changes, we should probably get it added to the wiki or on the documentation portal.
> > 
> > 
> > As these are mostly config files there's not a lot you can look out for. Looking out for typos like Niall did is pretty much it - links are being checked by docfx, so there would be a warning if I put a broken link in one of the toc files or md files.
> 
> Is there a mechanism for running docfx building locally?

yep. it's fairly easy - you download the zip package https://dotnet.github.io/docfx/ and either place the content of that in the folder you're running docfx from or the easier way, you add docfx.exe to your PATH, then you navigate to the root folder of your project where docfx.json resides (root folder of mrtk_development branch after this PR is finished) - and you call docfx docfx.json (optionally with the -f  force option to rebuild everything)

**Pull request merged**